pkg/config/parameters.go

```diff
@@ -383,6 +383,15 @@ type NetworkParameters struct {
 	XffNumTrustedHops uint32 `yaml:"num-trusted-hops"`
 }
 
+// ListenerParameters hold various configurable listener values.
+type ListenerParameters struct {
```

Comment:
Feels like these are really upstream settings, not listener settings. How about we call it something like `upstream`, or `upstreamConfig` or similar? @skriss @stevesloka @sunjayBhatia, any thoughts?

Reply:
I think the balancing is done on which Envoy worker thread will accept and do work on a connection, I would say tying "upstream" into the naming makes this seem like an upstream load balancing parameter which it isn't

Reply:
Maybe remove "Use" and have the name be something to the effect of "evenly distribute accepted connections between Envoy workers"

Reply:
let us read the envoy docs. there is some params:

 ```
 "name": "...",
  "address": "{...}",
  "filter_chains": [],
  "use_original_dst": "{...}",
  "per_connection_buffer_limit_bytes": "{...}",
  "metadata": "{...}",
  "drain_type": "...",
  "listener_filters": [],
  "listener_filters_timeout": "{...}",
  "continue_on_listener_filters_timeout": "...",
  "transparent": "{...}",
  "freebind": "{...}",
  "socket_options": [],
  "tcp_fast_open_queue_length": "{...}",
  "traffic_direction": "...",
  "udp_listener_config": "{...}",
  "api_listener": "{...}",
  "connection_balance_config": "{...}",
  "reuse_port": "...",
  "access_log": []
}
```

Exact-balance is one of the parameters, I think contour may be added to support other parameter settings in the future.

So i think  the listenterParams is appropriate

Reply:
That makes sense, I guess we're adding `listener` then.

I think we should call the inner parameter `connectionBalancer` and have the only valid value be `exact`. Other values should be ignored, with an error logged. Putting the wrong value should not be fatal.

Yes, I know, this is a more complicated way to do what's a boolean anyway, but this way it's more descriptive, shorter, and allows for more flexibility in the future.

So, the stanza in YAML should look like:
```yaml
listener:
  connectionBalancer: exact
```
What do you think? @sunjayBhatia @iyacontrol?